packages/ui/src/components/JobCard/JobCard.tsx

```diff
@@ -35,49 +40,82 @@ export const JobCard = ({
   jobUrl,
 }: JobCardProps) => {
   const { t } = useTranslation();
+  const { collapseJob } = useSettingsStore();
+
+  const [isOpen, setIsOpen] = React.useState<boolean>();
+
+  React.useEffect(() => {
+    if (isOpen === undefined)
+      setIsOpen(!jobUrl ? true : !collapseJob)
+  }), [];
```

Comment:
it definitelly works, try it :)

this is "onComponentMount" so the initial state is undefined:

- If we are on a job page, the card is instantly expanded, and then theres no collapse button.
- If we are on the list page, the card takes the value defined in the app setting and from them controlled by only the state

Reply:
It doesn't, since the array of deps is outside of the useEffect :P

Reply:
I reworked the logic to make it useEffect-free and handle the global state "live":
it default to the state defined globally, if theres any state change at job level, this is preserved, even if the global state changed, because now the job uses the local as the preferred:


https://github.com/felixmosh/bull-board/assets/23552631/0daaa98a-d189-4518-8be4-e6351fd6d86e